test/e2e/apps/controller_revision.go

```diff
@@ -0,0 +1,266 @@
+		err = wait.PollImmediate(controllerRevisionRetryPeriod, controllerRevisionRetryTimeout, checkControllerRevisionListQuantity(f, dsLabelSelector, 1))
+		framework.ExpectNoError(err, "failed to count required ControllerRevisions")
+
+		listControllerRevisions, err := csAppsV1.ControllerRevisions(ns).List(context.TODO(), metav1.ListOptions{})
+		currentControllerRevision := listControllerRevisions.Items[0]
+
+		ginkgo.By(fmt.Sprintf("Updating ControllerRevision %q", currentControllerRevision.Name))
+		var updatedControllerRevision *appsv1.ControllerRevision
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			updatedControllerRevision, err = csAppsV1.ControllerRevisions(ns).Get(context.TODO(), currentControllerRevision.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Unable to get ControllerRevision %s", currentControllerRevision.Name)
+			updatedControllerRevision.Labels[currentControllerRevision.Name] = "updated"
+			updatedControllerRevision, err = csAppsV1.ControllerRevisions(ns).Update(context.TODO(), updatedControllerRevision, metav1.UpdateOptions{})
+			return err
+		})
+		framework.ExpectNoError(err, "failed to update ControllerRevision in namespace: %s", ns)
+		framework.ExpectEqual(updatedControllerRevision.Labels[currentControllerRevision.Name], "updated", "Did not find 'updated' label for this ControllerRevision. Current labels: %v", currentControllerRevision.Labels)
+		framework.Logf("%s has been updated", updatedControllerRevision.Name)
+
+		ginkgo.By("Generate another ControllerRevision by patching the Daemonset")
+		patch := fmt.Sprintf(`{"spec":{"template":{"spec":{"terminationGracePeriodSeconds": %d}}},"updateStrategy":{"type":"RollingUpdate"}}`, 1)
+
+		_, err = c.AppsV1().DaemonSets(ns).Patch(context.TODO(), dsName, types.StrategicMergePatchType, []byte(patch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "error patching daemon set")
+
+		ginkgo.By("Confirm that there are two ControllerRevisions")
+		err = wait.PollImmediate(controllerRevisionRetryPeriod, controllerRevisionRetryTimeout, checkControllerRevisionListQuantity(f, dsLabelSelector, 2))
+		framework.ExpectNoError(err, "failed to count required ControllerRevisions")
+
+		updatedLabel := map[string]string{updatedControllerRevision.Name: "updated"}
+		updatedLabelSelector := labels.SelectorFromSet(updatedLabel).String()
+
+		ginkgo.By(fmt.Sprintf("Removing a ControllerRevision via 'DeleteCollection' with labelSelector: %q", updatedLabelSelector))
+		err = csAppsV1.ControllerRevisions(ns).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: updatedLabelSelector})
+		framework.ExpectNoError(err, "Failed to delete ControllerRevision: %v", err)
+
+		ginkgo.By("Confirm that there is only one ControllerRevision")
+		err = wait.PollImmediate(controllerRevisionRetryPeriod, controllerRevisionRetryTimeout, checkControllerRevisionListQuantity(f, dsLabelSelector, 1))
```

Comment:
could we also check the expected revision numbers?

Reply:
I've included a check at the end of the test to confirm the expected revision number.